### ISSUE TITLE
Accounts blocked for no use oficial Whatsapp App

### ISSUE THREAD

**Issue by inthemix84** [davidgfnet/whatsapp-purple#375]
Hi there

about 15 days ago, my account was suspended Whatsapp. I contacted support Whatsapp, and the reasons for which are suspended my account I have violated the terms of the contract, a customer using "unauthorized" for them, such as WhatsApp Purple Project.
Evidently they have way to detect the difference between the official client and an alternative, after returning to reactivate my account registration process from an "android" device and with an official version of Whatsapp, my account has been blocked again when logging Purple Whatsapp then with a few minutes.
I am using version 0.9 of the plugin.
It would be good to try to "simulate" the official client autentisidad to prevent future blockages.

Thank you

**Comment by Chamelleon** [comment]
Mine account also was blocked 2 minutes after turning on Pidgin.

**Comment by Docbroke** [comment]
mine blocked too. So now my both numbers are blocked. Last time I made mistake of changing android resource without asking for new password, but this time whatsapp knows that we are on unofficial application. Moving to telegram now, permenantly.

**Comment by davidgfnet** [comment]
Yeah sorry to hear that. I might add some extra features to mimic android better.
Actually Whatsapp sends some "stats" to their servers that might contain sensible data and doens't allow the user to disable that. Take that privacy!

**Comment by inthemix84** [comment]
Hi there

Thanks for all the answers, all the friends of the Purple Project. I hope soon to resolve the obstacles impeding further using Whatsapp Purple.

A big hello and good luck.

Nahuel

**Comment by stevesatriani** [comment]
I'll wait for update :smile: 
I am tired, Whatsapp always banned my account every 1 week
:cry:

**Comment by qualano** [comment]
I'm using the latest whatsapp-purple version. 

And I use Android-2.12.419 as resource.

Whatsapp-purple worked in the past but now i get: 

 root │ whatsapp - Logging in: Connecting
16:21 @        root │ whatsapp - Logging in: Awaiting response
16:21 @        root │ whatsapp - Login error: not-authorized
16:21 @        root │ whatsapp - Logging in: Awaiting response
16:21 @        root │ whatsapp - Logging in: Signing off..

Are they blocking all whatsapp-purple users?

Or should i try to register again with yowsup?

**Comment by inthemix84** [comment]
Hi there

To me I lock a month ago about Whatsapp service.

With Yousup and official Whatsapp I could not re-enlist the bill, I had to send an email to support Whatsapp to reactivate my account, which had been permanently locked.

Whatsapp support, answer me that my account was blocked temporarily by using an unauthorized (such as Whatsapp-Purple) client.
Until I send a screenshot of my mobile phone with the registration process with the official client.

Then, use a cell phone from a friend, to re-register and re-activate my account, sending the screenshot.

Once the support Whatsapp I found that the registration attempt from an authorized client, my mind returned to work.

At that time, use Yousup to get the key in plain text, and configure the plugin Purple.
The Purple Whatsapp-run for a few minutes with no problems receiving messages, but if you send.
After a few minutes, the account is again blocked, it is no longer possible to re-register with yousup, Whatsapp support me again say that my account was blocked by detecting the use of an unauthorized client.

Until fail to "simulate" the behavior of the official client, Whatsapp-Purple can hardly be used.

Cheers

**Comment by stukev** [comment]
@davidgfnet I have an impractical idea how to escape this problem. You could run Whatsapp web on a hidden browser and grab all data into pidgin from there. It's really bad practice but I've currently hacked together a more or less stable Whatsapp pidgin client this way.

I'm using AutoIT to spawn a hidden Firefox window, make it open Whatsapp web, monitor the page data, then send updates from and to a dummy plugin in Pidgin.

**Comment by davidgfnet** [comment]
@stukev Why don't you just use the web interface then? Most people here use this plugin cause they don't have a smartphone. Or they just don't want to use it.

**Comment by stukev** [comment]
@davidgfnet Plain aesthetic preference, I used 5 different messengers before and got fed up and was able to use Pidgin to replace 4 of them, whatsapp being the last exception as I too got banned when using whatsapp-purple. I didn't consider the no smartphone scenario, good luck with the reverse engineering.

**Comment by assegaf** [comment]
> Actually Whatsapp sends some "stats" to their servers that might contain sensible data and doens't allow the user to disable that. Take that privacy!

any idea how to send that stats to their server ? its just send one time after register or each login ?
